packages/flutter/lib/src/material/material_state.dart

```diff
@@ -152,16 +154,18 @@ abstract class MaterialStateColor extends Color {
 
   /// Returns a [Color] that's to be used when a Material component is in the
   /// specified state.
+  @override
   Color resolve(Set<MaterialState> states);
 
   /// Returns the color for the given set of states if `color` is a
-  /// [MaterialStateColor], otherwise returns the color itself.
+  /// [MaterialStateProperty], otherwise returns the color itself.
   ///
   /// This is useful for widgets that have parameters which can be [Color] or
-  /// [MaterialStateColor] values.
+  /// `MaterialStateProperty<Color>` values.
   static Color resolveColor(Color color, Set<MaterialState> states) {
-    if (color is MaterialStateColor) {
-      return color.resolve(states);
+    final dynamic dynamicColor = color;
+    if (dynamicColor is MaterialStateProperty<Color>) {
```

Comment:
Is there a reason to not just refer to and use `MaterialStateColor` in here?

Reply:
Since I created the `MaterialStateProperty` interface, I think all the type checks should just be for that interface. `MaterialStateColor` is just one convenient way to implement that interface.